File: MonProjet/src/main/java/com/wha/springmvc/model/CompteRemunere.java
package com.wha.springmvc.model;

import java.util.ArrayList;
import java.util.Date;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@DiscriminatorValue("COMPTE_REMUNERE")
public class CompteRemunere extends Compte {
	
	static private float taux = 2f; //taux d'intérêts fixé à 2%
	private float interets = 0;
	
	
	//construteur
	public CompteRemunere() {
	}
	
	public CompteRemunere(float solde, Date dateCreation, ArrayList<Operation> operations, float interets) {
		super(solde, dateCreation, operations);
		this.interets = interets;
	}

	//getters-setters
	public static float getTaux() {
		return taux;
	}
	
	public static void setTaux(float taux) {
		CompteRemunere.taux = taux;
	}

	public float getInterets() {
		return interets;
	}
	
	public void setInterets(float interets) {
		this.interets = interets;
	}
	
	
}

File: GestiBank/src/models/compteRemunere.ts
import { Operation } from './operation';
import { Compte } from './compte';

export class CompteRemunere extends Compte {
	
	static taux: number;
	interets: number;
	
	constructor(IBAN: string, idClient: number, solde:number, dateCreation: Date, operations: Operation[], interets: number){
		super(IBAN, idClient, solde, dateCreation, operations);
		this.interets = interets;
	}
}